src/Sylius/Bundle/ResourceBundle/Controller/StateMachineInterface.php

```diff
@@ -28,6 +29,10 @@ public function can(RequestConfiguration $configuration, ResourceInterface $reso
     /**
      * @param RequestConfiguration $configuration
      * @param ResourceInterface $resource
+     *
+     * @return bool
+     *
+     * @throws UpdateHandlingException
      */
-    public function apply(RequestConfiguration $configuration, ResourceInterface $resource): void;
+    public function apply(RequestConfiguration $configuration, ResourceInterface $resource): bool;
```

Comment:
It's a BC break, we can't change return typehints once we set them 🎉

Reply:
Thanks for your response pamil. If we have own SMinterface, I think it has to be the same as Winzou SMinterface - apply method return typehint is bool. Or is there any reason, why our apply method has void typehint?

Reply:
I guess it has never returned anything, so we added that `void` typehint to ensure that. Changing it right now would cause BC breaks for anyone reimplementing this interface in the userland code. I think instead of relying on this method returning `bool`, we can catch the exceptions outside of it and process them as a `false` :)

Reply:
Ok, I'll try to look at it. Thank you.

Reply:
@czende did you find the way to avoid this BC break?

Reply:
@Zales0123 Not quite yet :/ I have it dirty hardcoded in my project. To be sure, I have no clue how to avoid BC break. When you look at `SM\StateMachine\StateMachineInterface` both methods `can()` and `apply()` return `bool`, so I don't know why we have `bool` and `void` in `Sylius\Bundle\ResourceBundle\Controller\StateMachineInterface`.